### ISSUE TITLE
[EXE] idea for exercise 005 : Sentiment analysis

### ISSUE THREAD

**Issue by avats-dev** [gimseng/99-ML-Learning-Projects#68]
#### Learning Goals
- learn preprocessing of text data and various important processes such as tokenization and stemming
- learn regular expressions through stopwords removal from text
- learn logistic regression
- learn naive bayes
- learn application of neural networks on text data
- learn vectorization techniques and mathematical implementations through numpy
- learn and use libraries such as pandas, numpy, nltk, tensorflow, keras and scikit-learn.

### Exercise Statement
In this exercise, one would be able to know how to analyse sentiments of text data using various conventional and advanced algorithms along with textual data processing techniques.
- apply logistic reegression for sentiment analysis
- apply naive bayes for sentiment analysis
- apply word2vec algorithms

### Prerequisites 
This exercise goes from basic methods to advanced ones, so there are no hard requisites for this exercise. But it is recommended that one should know basic ML workflow to grasp things conveniently.

### Data source/summary:
Data has been taken from various sources and public datasets available.
> I would update this later.

### (Optional) Suggest/Propose Solutions
I have a solution and I would be adding a PR for the same. However, one is welcome:
- to enhance the existing solution
- to add other methods and algorithms for sentiment analysis

### (Optional) Further Links/Credits to Relevant Resources:
I would update this later with various references.

**Comment by gimseng** [comment]
@avats-dev great idea ! I vaguely remember there's a Kagggle twitter project for this. At any rate, go ahead and create a PR when you are ready. Many thanks !

**Comment by avats-dev** [comment]
Thanks @gimseng , I will do so. It might take some time as I'm trying to incorporate many methods in this one and to get a relative idea of varying accuracy across different models.

**Comment by gimseng** [comment]
@avats-dev Take your time. One comment though: Don't need to worry about covering comprehensive ideas/methods, if the exercise gets too big. Even though we are early in this repo, the idea is hopefully there will be incremental improvements to each exercise hopefully by different people at different times. Many thanks !

**Comment by tejasvi541** [comment]
I have Similar Sentiment Analysis Project with me which i have made watching a tutorial, using the amazon review data from jmcauley.uscd.edu/data/amazon. My model is not that perfect but you can use that data.

**Comment by avats-dev** [comment]
Hey @tejasvi541 , feel free to open a PR with your model, and then everyone else could review it.
> Btw I checked out the dataset that you have used and it's sufficient so that others can also improve it with better algorithms and models in future.

**Comment by gimseng** [comment]
@avats-dev @tejasvi541 Thanks for all the comments ! I will let both of you figure out the collaboration. Btw, if you want to have more detailed chat, feel free to join the discord. I created a '005-sentiment_analysis' chat channel inside the project section. So a less formal and chatty discussion could happen there. Feel free to use/not use it, so to let you know that its there. Looking forward to the PR! Again many thanks !
